### PR TITLE
BUG/REF: ArrowExtensionArray non-nanosecond units

### PR DESCRIPTION
**Pull request by lukemanley**
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/v2.1.0.rst` file if fixing a bug or adding a new feature.

Fixes a few bugs related to non-nanosecond units for pyarrow duration and timestamp types.

The underlying issue is upstream (https://github.com/apache/arrow/issues/33321) where pyarrow does not handle pandas `Timedelta` and `Timestamp` non-nano units correctly.

Example (1 seconds -> 0 seconds):

```
In [1]: import pandas as pd

In [2]: import pyarrow as pa

In [3]: td = pd.Timedelta(1, unit="s").as_unit("s")

In [4]: pa.scalar(td, type=pa.duration("s"))
Out[4]: <pyarrow.DurationScalar: datetime.timedelta(0)>

In [5]: pa.array([td], type=pa.duration("s"))
Out[5]: 
<pyarrow.lib.DurationArray object at 0x10751d120>
[
  0
]
```

### CODE REVIEW

File: pandas/core/arrays/arrow/array.py [hunk 1]
```diff
@@ -348,6 +321,152 @@ def _from_sequence_of_strings(
...
+        pa.Scalar
+        """
+        if isinstance(value, pa.Scalar):
+            pa_scalar = value
+        elif isna(value):
+            pa_scalar = pa.scalar(None, type=pa_type)
+        else:
+            # GH 53171: pyarrow does not yet handle pandas non-nano correctly
+            # see https://github.com/apache/arrow/issues/33321
+            from pandas._libs.tslibs import (
```
Comment (mroeschke) [inline pandas/core/arrays/arrow/array.py:365]: Is there a circular import if this is put at the top?

File: pandas/core/arrays/arrow/array.py [hunk 2]
```diff
@@ -348,6 +321,152 @@ def _from_sequence_of_strings(
...
+                value = value.copy()
+
+            if (
+                pa_type is not None
+                and pa.types.is_duration(pa_type)
+                and (not isinstance(value, np.ndarray) or value.dtype.kind not in "mi")
+            ):
+                # GH 53171: pyarrow does not yet handle pandas non-nano correctly
+                # see https://github.com/apache/arrow/issues/33321
+                from pandas.core.tools.timedeltas import to_timedelta
```
Comment (mroeschke) [inline pandas/core/arrays/arrow/array.py:437]: Likewise for these imports?

**Comment by jbrockmendel** [comment]
Is there some amount of this we can hope to rip out once pyarrow is updated?

File: pandas/core/arrays/arrow/array.py (hunk 1 above)
Comment (lukemanley) [reply pandas/core/arrays/arrow/array.py]: updated, this is now at the top

File: pandas/core/arrays/arrow/array.py (hunk 2 above)
Comment (lukemanley) [reply pandas/core/arrays/arrow/array.py:436]: `to_timedelta` results in a circular import if moved to the top.

**Comment by lukemanley** [comment]
> Is there some amount of this we can hope to rip out once pyarrow is updated?

I think it can all be removed eventually. It may take some time as it needs to be fixed upstream first and then the pyarrow min version needs to catch up.

**Review by mroeschke — COMMENTED** [review]
Looks okay. Just a merge conflict

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @lukemanley

**Pull request merged**